packages/core/lib/segments/segment.js

```diff
@@ -346,18 +339,30 @@ Segment.prototype.close = function(err, remote) {
 Segment.prototype.flush = function flush() {
   if (this.notTraced !== true) {
     delete this.exception;
-    SegmentEmitter.send(omit(this, ['counter', 'notTraced']));
+
+    var thisCopy = Utils.objectWithoutProperties(
+      this,
+      ['counter', 'notTraced'],
+      true
+    );
+
+    SegmentEmitter.send(thisCopy);
   }
 };
 
 Segment.prototype.format = function format() {
-  var ignore = ['counter', 'notTraced', 'exception'];
+  var ignore = ['segment', 'parent', 'counter'];
 
-  if (isEmpty(this.subsegments))
+  if(this.subsegments && this.subsegments.length === 0)
```

Comment:
`isEmpty` from Lodash [returns true](https://lodash.com/docs/#isEmpty) if passed a `null` object or `undefined` value, so please mimic that behavior in all cases of replacing isEmpty.

Reply:
Addressed in 5f9ab02.